canonical/cloud-init
# Ubuntu/bionic

Backport  cc_ubuntu_pro module to bionic to ensure all ESM-supported series allow using the `ubuntu_pro:` key in cloud-config user-data to align with product naming.

## do not merge. Merge will be performed on the cmdline


## Proposed Commit Message
```
    feat(pro): deprecate ubuntu_advantage key for ubuntu_pro
    
    Backport deprecatation of ubuntu_advantage_key in favor of ubuntu_pro
    to standardize product branding for Ubuntu Pro.
    
    This allows for common behavior, documentation and tooling on all
    
    Backport deprecatation of ubuntu_advantage_key in favor of ubuntu_pro
    to standardize product branding for Ubuntu Pro.
    
    This allows for common behavior, documentation and tooling on all
    supported and maintained releases including those Ubuntu releases
    under active Expanded Security Maintenance(ESM).
    
    LP: #2067660
```

## Additional Context
SRU process bug
[LP: #2067660](https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/2067600)

## Test Steps
```
lxc launch ubuntu-daily:bionic test-pro-bionic
lxc exec test-pro-bionic -- add-apt-repository ppa:chad.smith/esm-backport-ubuntu-pro-config -y
lxc exec test-pro-bionic -- apt-install cloud-init -y
cat > user-data <<EOF
#cloud-config
ubuntu_pro:
   token: <YOUR_TOKEN_FROM_http://ubuntu.com/pro/dashboard>
EOF
lxc file push user-data test-pro-bionic/var/lib/cloud/seed/nocloud-net/
lxc exec test-pro-bionic -- cloud-init clean --logs --reboot
lxc exec test-pro-bionic -- cloud-init status --wait --format=yaml  # ensure no errors
lxc exec test-pro-bionic -- pro status   # ensure attached
```
## Merge type
